wolfcrypt/src/port/silabs/silabs_ecc.c

```diff
@@ -0,0 +1,319 @@
+/* silabs_ecc.c
+ *
+ * Copyright (C) 2006-2020 wolfSSL Inc.
+ *
+ * This file is part of wolfSSL.
+ *
+ * wolfSSL is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * wolfSSL is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1335, USA
+ */
+
+#ifdef HAVE_CONFIG_H
+    #include <config.h>
+#endif
+
+
+#include <wolfssl/wolfcrypt/settings.h>
+
+#if defined(WOLFSSL_SILABS_SE_ACCEL)
+
+#include <wolfssl/wolfcrypt/error-crypt.h>
+#include <wolfssl/wolfcrypt/ecc.h>
+#include <wolfssl/wolfcrypt/port/silabs/silabs_ecc.h>
+
+
+
+sl_se_key_type_t silabs_map_key_type (ecc_curve_id curve_id)
+{
+    sl_se_key_type_t res = 0;
+
+    switch(curve_id) {
+    case ECC_SECP192R1:
+        res = SL_SE_KEY_TYPE_ECC_P192;
+        break;
+    case ECC_SECP256R1:
+        res = SL_SE_KEY_TYPE_ECC_P256;
+        break;
+
+#ifdef SL_SE_KEY_TYPE_ECC_P384
+    case ECC_SECP384R1:
+        res = SL_SE_KEY_TYPE_ECC_P384;
+        break;
+#endif
+
+#ifdef SL_SE_KEY_TYPE_ECC_P521
+    case ECC_SECP521R1:
+        res = SL_SE_KEY_TYPE_ECC_P521;
+        break;
+#endif
+
+#if defined(HAVE_CURVE25519) && defined(SL_SE_KEY_TYPE_ECC_X25519)
+    case ECC_X25519:
+        res = SL_SE_KEY_TYPE_ECC_X25519;
+        break;
+#endif
+
+#if defined(HAVE_CURVE448) && defined(SL_SE_KEY_TYPE_ECC_X448)
+    case ECC_X448:
+        res = SL_SE_KEY_TYPE_ECC_X448;
+        break;
+#endif
+
+    default:
+        // TODO: log error
```

Comment:
Recommend use the return code `NOT_COMPILED_IN`.

Reply:
Will return different values and check value.